arangod/RocksDBEngine/RocksDBMetaCollection.cpp

```diff
@@ -962,14 +970,6 @@ futures::Future<Result> RocksDBMetaCollection::rebuildRevisionTree() {
           std::chrono::milliseconds(RandomGenerator::interval(uint32_t(2000))));
     }
 
-    {
-      // we may still have some buffered updates, so let's remove them.
-      std::unique_lock<std::mutex> guard(_revisionTreeLock);
-      removeBufferedUpdatesUpTo(blockerSeq - 1);
-
-      _revisionTreeApplied = blockerSeq - 1;
-    }
```

Comment:
Removing buffered updates will break the existing merkel tree, if the replacement procedure fails later on. For example due to a timeout.